### PR TITLE
Refinery/MarkdownToHtml: Escape HTML and don't accept and unsafe links

### PR DESCRIPTION
**Pull request by mjansenDatabay**
See: https://mantis.ilias.de/view.php?id=40204

If approved, this has to be picked to `trunk` as well.

### CODE REVIEW

**Comment by corro** [comment]
Maybe consider setting `max_nesting_level` too in order to avoid excessively long rendering times and possible DoS.

File: tests/Refinery/String/MarkdownFormattingToHTMLTest.php [hunk 1]
```diff
@@ -40,6 +39,7 @@ protected function setUp(): void
         $group = new Group(new Factory(), $language);
 
         $this->markdown = $group->markdown()->toHTML();
+        $this->markdown_wihtout_html = $group->markdown(true)->toHTML();
```
Comment (klees) [inline tests/Refinery/String/MarkdownFormattingToHTMLTest.php]:
```suggestion
        $this->markdown_without_html = $group->markdown(true)->toHTML();
```

File: src/Refinery/String/MarkdownFormattingToHTML.php [hunk 2]
```diff
@@ -29,9 +29,17 @@ class MarkdownFormattingToHTML
 {
     private \League\CommonMark\CommonMarkConverter $converter;
 
-    public function __construct()
+    public function __construct(bool $strip_html = false)
```
Comment (klees) [inline src/Refinery/String/MarkdownFormattingToHTML.php]: Lets default this to `true` so people would need to allow HTML willingly.

**Review by klees — CHANGES_REQUESTED** [review]
LGTM, please make the two changes. Feel free proceed afterwards. Thanks!

**Comment by mjansenDatabay** [comment]
@klees I applied the requested changes.

Furthermore, I added the `max_nesting_level` configuration option (it is not configurable by our users), this was also a recommendation. I did not find a meaningful value for that (so I chose `42`, maybe you or @chfsx have any serious suggestion here, currently it is just a "Magic Number"), the documentation suggests a value between 10 and 50.

According to the suggestion of @chfsx I changed the HTML handling to `escape` instead of `strip`.

**Review by klees — APPROVED** [review]

**Pull request merged**